jaxtyping/_decorator.py

```diff
@@ -33,10 +33,17 @@
     get_type_hints,
     NoReturn,
     overload,
-    ParamSpec,
     TypeVar,
 )
 
+
+try:
+    from typing import ParamSpec
+except ImportError:
+    # Python < 3.10
+    from typing_extensions import ParamSpec
```

Comment:
FYI https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-new-additions-to-the-typing-module

```python
import sys
if sys.version_info < (3, 10):
    from typing_extensions import ParamSpec
else:
    from typing import ParamSpec
```

Reply:
Do you want me to make another PR with this way of doing it?

Reply:
No strong feelings from me! :)

Reply:
Maybe next time. 😊 The semantics are different, but the main thing is that there's an import guard.

I was vendoring [jaxtyping](https://aur.archlinux.org/packages/python-jaxtyping) in Arch and noticed a dependency on `typing-extensions`. Since Arch is rolling release distro, Python is the latest and `typing-extension` dependency usually can be omitted. It is quite common mistake to import unconditionally `typing_extensions` or `importlib.metadata`.